src/core/runner.ts

```diff
@@ -451,11 +451,11 @@ export default class Runner {
       env: options.environment,
       params: options.params,
     });
-    await this.reset();
+    await this.reset(options);
     return result;
   }
 
-  async reset() {
+  async reset(options?: RunOptions) {
```

Comment:
[just a point of view]

I don't tend to be a fan of passing args in cascade. For instance, `reset` doesn't need `dry-run` flag, but now it "depends" on it.

The way I see it is if a reporter behaves differently depending on something (in this case the dry-run flag) I would add that something as a reporter option: https://github.com/elastic/synthetics/blob/main/src/reporters/base.ts#L76 and https://github.com/elastic/synthetics/blob/main/src/core/runner.ts#L155, so everything stays in the "reporter realm"

With that:

- no more cascade args passing
- If other methods within a reporter need to behave differently because of a dry-run, no need to change their signature.